allennlp/predictors/bidaf.py

```diff
@@ -38,3 +44,101 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
         question_text = json_dict["question"]
         passage_text = json_dict["passage"]
         return self._dataset_reader.text_to_instance(question_text, passage_text)
+
+    @overrides
+    def predictions_to_labeled_instances(self,
+                                         instance: Instance,
+                                         outputs: Dict[str, np.ndarray]) -> List[Instance]:
+        """
+        NAQANET has the following fields, answer_as_passage_spans,  answer_as_question_spans,
+        answer_as_add_sub_expressions, answer_as_counts. We need to provide labels for all of them.
+        """
+        new_instance = deepcopy(instance)
+        # For BiDAF
+        if 'best_span' in outputs:
+            span_start_label = outputs['best_span'][0]
+            span_end_label = outputs['best_span'][1]
+            new_instance.add_field('span_start',
+                                   IndexField(int(span_start_label), new_instance['passage'])) # type: ignore
+            new_instance.add_field('span_end',
+                                   IndexField(int(span_end_label), new_instance['passage'])) # type: ignore
+
+        # For NAQANet model
+        elif 'answer' in outputs:
+            answer_type = outputs['answer']['answer_type']
+
+            # When the problem is a counting problem
+            if answer_type == 'count':
+                field = ListField([LabelField(int(outputs['answer']['count']), skip_indexing=True)])
+                new_instance.add_field('answer_as_counts', field)
+
+            # When the answer is in the passage
+            elif answer_type == 'passage_span':
+                span = outputs['answer']['spans'][0]
+
+                # Convert character span indices into word span indices
+                word_span_start = None
+                word_span_end = None
+                offsets = new_instance['metadata'].metadata['passage_token_offsets'] # type: ignore
+                for idx, offset in enumerate(offsets):
+                    if offset[0] == span[0]:
+                        word_span_start = idx
+                    if offset[1] == span[1]:
+                        word_span_end = idx
+
+                passage_field: SequenceField = new_instance['passage']  # type: ignore
+                field = ListField([SpanField(word_span_start,
+                                             word_span_end,
+                                             passage_field)]) # type: ignore
```

Comment:
Why do you need a `type: ignore` here?  I think you can remove it from this line.